Firestore/core/test/firebase/firestore/local/remote_document_cache_test.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_TEST_FIREBASE_FIRESTORE_LOCAL_REMOTE_DOCUMENT_CACHE_TEST_H_
+#define FIRESTORE_CORE_TEST_FIREBASE_FIRESTORE_LOCAL_REMOTE_DOCUMENT_CACHE_TEST_H_
+
+#include <memory>
+
+#include "absl/strings/string_view.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+class Document;
+
+}  // namespace model
+
+namespace local {
+
+class Persistence;
+class RemoteDocumentCache;
+
+using FactoryFunc = std::unique_ptr<Persistence> (*)();
+
+/**
+ * These are tests for any implementation of the FSTRemoteDocumentCache
```

Comment:
Drop "FST" here and below (and replace "protocol").

Reply:
Done.